frontend/src/metabase/visualizations/visualizations/List.tsx

```diff
@@ -171,27 +172,56 @@ export default Object.assign(ListViz, {
 
     settings["link_text"] = {
       title: t`Link text`,
-      widget: "input",
+      widget: ChartSettingInputSuggestion,
       hint: linkFieldsHint,
       default: null,
       getHidden: (_: unknown, settings: VisualizationSettings) =>
         settings["view_as"] !== "link" && settings["view_as"] !== "email_link",
       readDependencies: ["view_as"],
-      props: {
-        placeholder: t`Link to {{bird_id}}`,
+      getProps: (
+        col: Column,
+        settings: VisualizationSettings,
+        onChange: (col: Column) => void,
+        {
+          series: [
+            {
+              data: { cols },
+            },
+          ],
+        }: { series: Series },
+      ) => {
+        return {
+          keys: cols.map(column => column.name),
```

Comment:
As stated in another comment. I think `keys` could look confusing as it's too generic in this case. I was confusing this with the key press we type in the input.

Maybe could be just a simple `columnNames` or `columnKeys` (I think I've seen `columnKey` somewhere)